ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/extensions/PaywallDataExtensions.kt

```diff
@@ -0,0 +1,100 @@
+package com.revenuecat.purchases.ui.revenuecatui.extensions
+
+import androidx.compose.material.Colors
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.toArgb
+import com.revenuecat.purchases.Package
+import com.revenuecat.purchases.paywalls.PaywallColor
+import com.revenuecat.purchases.paywalls.PaywallData
+import com.revenuecat.purchases.ui.revenuecatui.data.processed.PaywallTemplate
+import java.net.URL
+import java.util.Locale
+
+/***
+ * Default [PaywallData] to display when attempting to present a [PaywallView] with an offering that has no paywall
+ * configuration, or when that configuration is invalid.
+ */
+internal fun PaywallData.Companion.createDefault(
+    packages: List<Package>,
+    packageName: String,
+    currentColors: Colors,
+): PaywallData {
+    return PaywallData.createDefaultForIdentifiers(packages.map { it.identifier }, packageName, currentColors)
+}
+
+private fun PaywallData.Companion.createDefaultForIdentifiers(
+    packageIdentifiers: List<String>,
+    packageName: String,
+    currentColors: Colors,
+): PaywallData {
+    return PaywallData(
+        templateName = PaywallData.defaultTemplate.id,
+        config = PaywallData.Configuration(
+            packages = packageIdentifiers,
+            images = PaywallData.Configuration.Images(
+                background = PaywallData.defaultBackgroundImage,
+                icon = PaywallData.defaultAppIconPlaceholder,
+            ),
+            colors = PaywallData.defaultColors(currentColors),
+            blurredBackgroundImage = true,
+            displayRestorePurchases = true,
+        ),
+        localization = mapOf(Locale.ENGLISH.toString() to PaywallData.defaultLocalization),
+        assetBaseURL = PaywallData.defaultTemplateBaseURL(packageName),
+        revision = PaywallData.revisionID,
+    )
+}
+
+// region Internal defaults
+
+internal val PaywallData.Companion.defaultTemplate: PaywallTemplate
+    get() = PaywallTemplate.TEMPLATE_2
+
+internal val PaywallData.Companion.defaultAppIconPlaceholder: String
+    get() = "revenuecatui_default_paywall_app_icon"
+
+// endregion
+
+// region Private defaults
+
+private val PaywallData.Companion.revisionID: Int
+    get() = -1
+
+private val PaywallData.Companion.defaultBackgroundImage: String
+    get() = "R.drawable.default_background"
```

Comment:
So then, it will work with `RemoteImage` when using a local resource for the background? Checking just to make sure.

Reply:
I will address the background image stuff in a future PR because it requires some refactoring and changing URL to URIs

https://linear.app/revenuecat/issue/PWL-274/support-default-background-image